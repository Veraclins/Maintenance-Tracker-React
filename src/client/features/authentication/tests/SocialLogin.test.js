import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import SocialLogin from '../social-login/SocialLogin';

describe('Test the login presentation component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<SocialLogin />);
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without errors', () => {
    const wrapper = shallow(<SocialLogin />);
    expect(wrapper.find('div')).toHaveLength(5);
    expect(wrapper.find(Link)).toHaveLength(3);
  });
});
