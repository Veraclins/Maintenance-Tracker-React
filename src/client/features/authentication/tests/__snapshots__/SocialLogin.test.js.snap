// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the login presentation component renders without crashing 1`] = `
<div
  className="social col-md-6 pt-4"
>
  <br />
  <div
    className="text-center mb-4"
  >
    <em>
      Or connect with your favourite social media account
    </em>
  </div>
  <div
    className="text-center mb-1"
  >
    <Link
      replace={false}
      to="/oauth2/facebook"
    >
      <img
        alt="logo"
        className="mb-2 mt-2"
        height="59"
        src="Button_Facebook.svg"
        width="250"
      />
    </Link>
  </div>
  <div
    className="text-center mb-1"
  >
    <Link
      replace={false}
      to="/login"
    >
      <img
        alt="logo"
        className="mb-2 mt-2"
        height="59"
        src="Button_Twitter.svg"
        width="250"
      />
    </Link>
  </div>
  <div
    className="text-center mb-1"
  >
    <Link
      replace={false}
      to="/oauth2/google"
    >
      <img
        alt="logo"
        className="mb-2 mt-2"
        height="59"
        src="Button_Google.svg"
        width="250"
      />
    </Link>
  </div>
</div>
`;
